asapdiscovery-ml/asapdiscovery/ml/config.py

```diff
@@ -21,6 +23,29 @@
 from pydantic import BaseModel, Field, root_validator
 
 
+class ConfigBase(BaseModel):
+    """
+    Base class to provide update functionality.
+    """
+
+    def update(self, config_updates={}) -> ConfigBase:
```

Comment:
Misleading name as it doesn't update in place no? Returns a new updated object as far as I can see?

Reply:
Im not exactly sure what I would replace it with though, ignore if you can't think of something else.

Reply:
you definitely have a point, I was stuck on a bug for a while because I forgot that it wasn't an in-place update and assumed it was. I'll think on a replacement (or maybe just make it in place?)

Reply:
ok @hmacdope I think I'm going to leave it as is for now since it matches how I have it in `mtenn`, but I'll make an issue to change this in both repos. I think `new_from_update` instead of just `update` might make sense as the new name, but let me know what you  think

Reply:
Sounds good.